### ISSUE TITLE
recommendation logic doesn't work for specific inputs

### ISSUE THREAD

**Issue by martonsereg** [banzaicloud/telescopes#131]
to reproduce, run this `curl` against a local server:
```
curl 'http://localhost:9092/api/v1/recommender/ec2/eu-west-1/cluster/' --data-binary '{"sumCpu":12,"sumMem":8,"minNodes":1,"maxNodes":1,"onDemandPct":100}'
```

in this case the current recommendation flow looks like this:
1. cpu based: 12 cpu requested and 1 node min/max, so only **12 core machines** are selected
2. there's no such machine on AWS, so skip and go to memory based rec.
3. mem based: 8 gb memory machines are selected, and the filter is to have at least 12 cores.
4. there's no such machines either, so skip and fail the recommendation

We should still be able to provide a recommendation: the closest machines with at least these properties are `c5.4xlarge` that has 16 cpus and 32 gb ram, or `m5.4xlarge` with 16/64.

Will be fixed by: #135

**Comment by lpuskas** [comment]
To address this problem we might consider 2 approaches:
* introduce the notion of `deviance` - so that we allow a certain amount (expressed as percentage) of difference from the requested attribute values
* introduce strategies to follow; eg.: chose the next available value (higher or less) during recommendation

**Comment by lpuskas** [comment]
fixes #135
